lib/dor/models/itemizable.rb

```diff
@@ -23,20 +23,19 @@ def clear_diff_cache
 
     # Retrieves file difference manifest for contentMetadata from SDR
     #
-    # @param [String] subset keyword for file attributes :shelve, :preserve, :publish. Default is :all.
+    # @param [Symbol] subset keyword for file attributes :shelve, :preserve, :publish. Default is :all.
     # @param [String] version
-    # @return [String] XML contents of cm_inv_diff manifest
+    # @return [Moab::FileInventoryDifference] XML contents of cm_inv_diff manifest
     def get_content_diff(subset = :all, version = nil)
       if Dor::Config.stacks.local_workspace_root.nil?
         raise Dor::ParameterError, 'Missing Dor::Config.stacks.local_workspace_root'
       end
 
-      current_content = datastreams['contentMetadata'].content
-      if current_content.nil?
-        raise Dor::Exception, 'Missing contentMetadata datastream'
-      end
+      # If we don't have a contentMetadata datastream, return an empty difference.
+      # The technical-metadata robots relies on this functionality.
+      return Moab::FileInventoryDifference.new if datastreams['contentMetadata'].nil?
 
-      Sdr::Client.get_content_diff(pid, current_content, subset, version)
+      Sdr::Client.get_content_diff(pid, datastreams['contentMetadata'].content, subset, version)
     end
```

Comment:
here's a possible example for catching that exception -- is that what you had in mind?
```
    def self.get_content_group_diff(dor_item)
      inventory_diff = begin
          dor_item.get_content_diff('all')
        rescue Dor::Exception # no contentMetadata
          FileInventoryDifference.new
        end
      inventory_diff.group_difference('content')
    end
```